### ISSUE TITLE
Inconsistent behaviour between nbody and cbs drivers

### ISSUE THREAD

**Issue by PeterKraus** [psi4/psi4#977]
Mentioned in PR #975 .

The behaviour between the two equivalent CBS modes is inconsistent when passed through nbody driver:

```
molecule ne2 {
    0 1
    Ne 0 0 0
    --
    0 1
    Ne 0 0 2
}

set {
    scf_type      df
    mp2_type      df
    cc_type       df
    e_convergence 7
    reference     rhf
}

E_method_basis = energy("MP2/cc-pv[tq]z + D:CCSD(T)/cc-pv[dt]z", bsse_type="nocp")
E_cbs          = energy(cbs, bsse_type="nocp", corl_wfn="mp2", corl_basis="cc-pv[tq]z",
                                               delta_wfn="ccsd(t)", delta_basis="cc-pv[dt]z")

compare_values(E_method_basis, E_cbs, 6, "Interaction energy")   #TEST 
``` 

In the first case, `name = "MP2/[...]"` so the sequence is:
```
energy() -> _nbody_gufunc() -> [-> _cbs_gufunc() -> cbs()] * n => recurse back
```
In the second case, `name = cbs` which has `__call__`, so:
```
energy() -> cbs() -> [-> energy() -> _nbody_gufunc()] * x => recurse back
```
I am not sure what the best way of solving this is: 
- option 1) is to extend the n-body driver to accept `__call__`-able `name`s, 
- option 2) is to bounce out of cbs() when `bsse_type` is passed in, but this would require generating a string-based representation of the cbs method anyway

I don't mind having a go at this once I know which is the preferred solution!

**Comment by dgasmith** [comment]
If I understand it correctly both should give the same result. It is a matter of which is more optimal in the long run I think?

My gut says hitting nbody gufunc first is best as this splits everything into a large number of computations that can be parallelized easily.

**Comment by PeterKraus** [comment]
I agree that extending `_nbody_gufunc()` is probably the easier way to do this. I'll have a look at it next week.
